src/dds/dds-device-watcher.cpp

```diff
@@ -183,3 +177,17 @@ void dds_device_watcher::init( int domain_id )
 
     LOG_DEBUG( "DDS device watcher initialized successfully" );
 }
+
+
+bool dds_device_watcher::foreach_device(
+    std::function< bool( dds::dds_guid const &, std::shared_ptr< dds::dds_device > const & ) > fn )
+{
+    std::lock_guard< std::mutex > lock( _devices_mutex );
+    for( auto && guid_to_dev_info : _dds_devices )
+    {
+        if( ! fn( guid_to_dev_info.first, guid_to_dev_info.second ) )
+            return false;
+    }
+    return true;
```

Comment:
Same about the return value.
Why would we want to stop in a middle of the loop?
If we want to save it on the capture list as reference?  I cant think of another use case for it..

Reply:
👍